File: eCommerce/eCommerce.Services/Database/CuisineType.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace eCommerce.Services.Database
{
    public class CuisineType
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;
        
        [MaxLength(200)]
        public string? Description { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        // Navigation properties
        public ICollection<Restaurant> Restaurants { get; set; } = new List<Restaurant>();
        public ICollection<UserPreference> UserPreferences { get; set; } = new List<UserPreference>();
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/cuisine_type.dart
class CuisineType {
  final int id;
  final String name;
  final String? description;
  final bool isActive;

  CuisineType({
    this.id = 0,
    this.name = '',
    this.description,
    this.isActive = true,
  });

  factory CuisineType.fromJson(Map<String, dynamic> json) {
    return CuisineType(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'],
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'isActive': isActive,
    };
  }
}
